### ISSUE TITLE
How can we optimize GPU usage or transition from DDPShardedStrategy to FSDP?

### ISSUE THREAD

**Issue by fatbrowncown** [Doubiiu/DynamiCrafter#95]
Hi there,

I just wanted to express my appreciation for your incredible work. As I dive deeper into this field, I've encountered some limitations with my hardware resources - 2 GPU T4s, each with 16GB of VRAM. Currently, I'm trying to train a demo interpolation model using the first 100 videos of the WebVid-10M-motion dataset. To optimize my training, I made several adjustments:
+ Set accumulate_grad_batches to 8
+ Configured validation to occur every 10 epochs (check_val_every_n_epoch)
+ Implemented the DeepSpeed Stage 2 strategy

However, these adjustments don't seem to be sufficient, as I continue to encounter OOM errors. I'm seeking advice on further reducing GPU usage. Specifically:
1. Are there any additional parameters or strategies I can tweak?
2. If not, how can I transition from Distributed Data Parallel (DDP) to Fully Sharded Data Parallel (FSDP) to better manage memory consumption?

Thank you for your guidance. Have a nice day

**Comment by fatbrowncown** [comment]
I apologize, it seems FSDP is a data parallel strategy as well. So, I'll revise the second question to ask if **there's a way to switch from data parallel to model parallel**. Thank you really much!

**Comment by Doubiiu** [comment]
Hi. Thanks for your interest. The adjustments you've listed are all the ways I could think of... I am sorry that I cannot provide other useful suggestions. Hope to get good news from you.

**Comment by fatbrowncown** [comment]
Nah, never mind, just a pop up question ^^ I’ve upgraded my hardware to an A100 40GB, and it seems to be working fine with the GPU consuming around 39.5 GB of VRAM. Is this normal?
